src/Symfony/Component/Serializer/Mapping/ClassMetadata.php

```diff
@@ -78,11 +78,7 @@ public function merge(ClassMetadataInterface $classMetadata): void
 
     public function getReflectionClass(): \ReflectionClass
     {
-        if (!$this->reflClass) {
-            $this->reflClass = new \ReflectionClass($this->getName());
-        }
-
-        return $this->reflClass;
+        return $this->reflClass ??= new \ReflectionClass($this->getName());
```

Comment:
I'd suggest one more change:
```
private \ReflectionClass $reflClass;
```

Reply:
I had a discussion about this recently in the static analysis channel of Symfony Slack. It is currently impossible to document (either natively or through PHPdocs) that a property is not initialized on construction. This makes using uninitialized properties for optional properties not a great choice for the readability and usability of the code.

Unless there are performance improvements of uninitialized properties vs nullable ones, I would be in favor to keep using nullable types to indicate that a property is optional.

Reply:
An other angle to this: keeping the property uinitialized makes it clearer that it should be accessed directly, whearas accepting `null` says: it's OK to deal with null values when accessing this property.
I'm more on the uinitialized side for this reason.

Reply:
there is no clear way to mark a property as being lazily initialized (either natively or in PHPdoc). so keeping a property uninitialized in the constructor actually does not make it clear that access it requires special care (if it is nullable, you have to take care of null, which might make you use the getter method that has a non-nullable return type).
Note that static analysis tools don't help there, as none of them deal with uninitialized properties correctly: either they pretend they don't exist or they forbid them.